### PR TITLE
Added WebP support

### PR DESCRIPTION
**Pull request by stayawakesteve**
I've largely plagiarised from our old image tag helper from a while ago. Not too keen on having to loop through the sizes twice (as per the old image tag helper), so open to suggestions on how that could be made better 😄 Or ignore if it's fine as is!

### CODE REVIEW

File: Our.Umbraco.TagHelpers/ImgTagHelper.cs [hunk 1]
```diff
@@ -314,12 +314,63 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
...
+                            OurScreenSize.Medium => _globalSettings.OurImg.MediaQueries.Medium,
+                            OurScreenSize.Small => _globalSettings.OurImg.MediaQueries.Small,
+                            _ => 0
+                        };
+
+                        double sourceHeight = 0;
+
+                        if (MediaItem != null)
+                        {
+                            #region Configure crops which can be set at variant level or inherit from the crop alias defined on the main img element itself. If neither have a crop alias, then don't use crops.
```
Comment (peterkeating) [inline Our.Umbraco.TagHelpers/ImgTagHelper.cs:347]: I would remove this and make it a comment as the file doesn't use `#region` elsewhere.

File: Our.Umbraco.TagHelpers/ImgTagHelper.cs [hunk 2]
```diff
@@ -314,12 +314,63 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
...
+            var renderWebP = imageFormat != "gif";
+
             if (imageSizes != null && imageSizes.Any())
             {
                 var sb = new StringBuilder();
                 sb.AppendLine("<picture>");
 
                 imageSizes = _globalSettings.OurImg.MobileFirst ? imageSizes.OrderByDescending(o => o.ScreenSize).ToList() : imageSizes.OrderBy(o => o.ScreenSize).ToList();
+
+                if (renderWebP)
```
Comment (peterkeating) [inline Our.Umbraco.TagHelpers/ImgTagHelper.cs:329]: I'd suggest moving everything in this `if` statement be moved to it's own method? Move the `renderWebP` check logic in to the method too.

```
private void RenderWebP(args) {
    if (!renderWebP) {
        return;
    }

    ...
}
```

File: Our.Umbraco.TagHelpers/ImgTagHelper.cs [hunk 3]
```diff
@@ -314,12 +314,63 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // The configuration allows us to define whether images are configured "mobile first". This simply alternates between min-width & max-width media queries.
             var imageSizes = GetImageSizes(MediaItem != null);
 
+            // Avoid rendering a WebP alternative if the image is a GIF
+            var imageFormat = MediaItem != null ? MediaItem.Url().Split('.').Last() : FileSource?.Split('.').Last();
```
Comment (peterkeating) [inline Our.Umbraco.TagHelpers/ImgTagHelper.cs]: Assume this is getting the file extension, there is `Path.GetExtension` that removes the need for a bespoke approach.

File: Our.Umbraco.TagHelpers/ImgTagHelper.cs [hunk 4]
```diff
@@ -314,12 +314,63 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // The configuration allows us to define whether images are configured "mobile first". This simply alternates between min-width & max-width media queries.
             var imageSizes = GetImageSizes(MediaItem != null);
 
+            // Avoid rendering a WebP alternative if the image is a GIF
+            var imageFormat = MediaItem != null ? MediaItem.Url().Split('.').Last() : FileSource?.Split('.').Last(); 
+            var renderWebP = imageFormat != "gif";
```
Comment (peterkeating) [inline Our.Umbraco.TagHelpers/ImgTagHelper.cs]: This should probably be case insensitive, use `imageFormat.Equals("gif", StringComparison.OrdinalIgnoreCase)`.

**Review by peterkeating — COMMENTED** [review]
Looks good all in all. My suggestions are more to neaten things up.

File: Our.Umbraco.TagHelpers/ImgTagHelper.cs (hunk 1 above)
Comment (stayawakesteve) [reply Our.Umbraco.TagHelpers/ImgTagHelper.cs:347]: Good point, although this was in the original repo so will leave it as is. Not sure why it's being marked as changed.

**Pull request closed without merging**